### PR TITLE
nydus-image: add unit test for setting default compression algorithm

### PR DESCRIPTION
**Pull request by xwb1136021767**
## Relevant Issue (if applicable)
Add unit test for da501f758e3c0b5754b37cf8b5d5fd32b3ad4470.
Upgrade openssl to 0.10.55 to avoid RUSTSEC-2023-0044(https://rustsec.org/advisories/RUSTSEC-2023-0044.html).
## Details
_Please describe the details of PullRequest._

## Types of changes

_What types of changes does your PullRequest introduce? Put an `x` in all the boxes that apply:_

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation Update (if none of the other choices apply)

## Checklist

_Go over all the following points, and put an `x` in all the boxes that apply._

- [ ] I have updated the documentation accordingly.
- [x] I have added tests to cover my changes.

### CODE REVIEW

**Review by jiangliu — APPROVED** [review]

**Comment by jiangliu** [comment]
Please help to open a PR for master branch too,
Also please `cargo fmt` your changes.

File: src/bin/nydus-image/core/blob.rs [hunk 1]
```diff
@@ -260,3 +265,45 @@ impl Blob {
...
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    #[test]
+    fn test_default_compression_algorithm_for_meta_ci() {
+        let mut ctx = BuildContext::default();
+
+        //TarToRef
```
Comment (imeoer) [inline src/bin/nydus-image/core/blob.rs:277]: Maybe add a space after `//`.

File: src/bin/nydus-image/core/blob.rs [hunk 2]
```diff
@@ -123,6 +123,14 @@ impl Blob {
         Ok(())
     }
 
+    fn get_compression_algorithm_for_meta(ctx: &BuildContext) -> compress::Algorithm {
```
Comment (imeoer) [inline src/bin/nydus-image/core/blob.rs:126]: How about `fn get_meta_compressor`.

File: src/bin/nydus-image/core/blob.rs [hunk 3]
```diff
@@ -260,3 +265,45 @@ impl Blob {
...
+        ctx = BuildContext {
+            conversion_type: ConversionType::TargzToRef,
+            ..ctx
+        };
+        let compressor = Blob::get_compression_algorithm_for_meta(&ctx);
+        assert_eq!(compressor, compress::Algorithm::Zstd);
+
+        //TarToRef
+        ctx = BuildContext {
+            conversion_type: ConversionType::TarToRef,
```
Comment (imeoer) [inline src/bin/nydus-image/core/blob.rs:303]: Should also add a case for !`ConversionType::XXXToRef` ?

Comment (xwb1136021767) [reply src/bin/nydus-image/core/blob.rs:304]: > Should also add a case for !`ConversionType::XXXToRef` ?

Thank you for your suggestion, I have made the modifications according to your feedback, and then I will send a PR for the master.

File: Cargo.toml [hunk 4]
```diff
@@ -53,7 +53,7 @@ vmm-sys-util = "0.10.0"
 xattr = "0.2.3"
 
 # Build static linked openssl library
-openssl = { version = "0.10.48", features = ["vendored"] }
+openssl = { version = "0.10.55", features = ["vendored"] }
 # pin openssl-src to bring in fix for https://rustsec.org/advisories/RUSTSEC-2022-0032
```
Comment (adamqqqplay) [inline Cargo.toml:57]: Maybe you could split the openssl changes into another commit
